peekingduck/pipeline/nodes/abstract_node.py

```diff
@@ -155,3 +171,9 @@ def _edit_config(
                         f"Config for node {self.node_name} is updated to: '{key}': {value}"
                     )
         return dict_orig
+
+    def _get_config_types(self) -> Dict[str, Any]:
+        """Returns a dictionary which maps the node's config keys to their
+        respective typing.
+        """
+        return {}
```

Comment:
> Would be good to use the `@abstractmethod` decorator on `_get_config_types()` ...

`@abstractmethod` decorator enforces all child classes to implement the method, even if the nodes don't have any configs. Such as `dabble.bbox_count`, `draw.poses`, etc. These nodes will then need to have an empty `_get_config_types()` method.

Similarly, it will also require custom nodes to implement the method as well. The current implementation make this an optional feature instead of forcing users to adopt our way of type checking.

Would it still be preferable to enforce all child classes to implement a `_get_config_types()` with `@abstractmethod` decorator?

> ... add documentation on why `_get_config_types()` is to be called in `node.run()` and not in `node.__init__()`

For the child classes which implemented `_get_config_types()`, it's placed below the `run()` method. 

In `AbstractNode`, it's called in `load_node_config()` which is called in `AbstractNode.__init__()`.

Reply:
That's a good point: `_get_config_types()` may not always be required, unlike `run()`.
Then let's keep the current design.